### PR TITLE
Unify remaining CoreLib interop files

### LINKED ISSUE #1232: Unify corelib interop files with common ones
As an artefact of the repo merge we have files duplicated between src\libraries\Common\src\Interop and src\libraries\System.Private.CoreLib\src\Interop

The list is these (just going by file name)
```
Interop\Unix\Interop.Libraries.cs
Interop\Unix\System.Native\Interop.GetPid.cs
Interop\Windows\Interop.Errors.cs
Interop\Windows\Interop.Libraries.cs
Interop\Windows\Kernel32\Interop.Constants.cs
Interop\Windows\Kernel32\Interop.CreateFile.cs
Interop\Windows\Kernel32\Interop.FILE_INFO_BY_HANDLE_CLASS.cs
Interop\Windows\Kernel32\Interop.FreeLibrary.cs
Interop\Windows\Kernel32\Interop.GetCurrentProcessId.cs
Interop\Windows\Kernel32\Interop.GetFileInformationByHandleEx.cs
Interop\Windows\Kernel32\Interop.GetProcessTimes.cs
Interop\Windows\Kernel32\Interop.GlobalMemoryStatusEx.cs
Interop\Windows\Kernel32\Interop.LoadLibraryEx.cs
Interop\Windows\Kernel32\Interop.LocalAlloc.cs
Interop\Windows\Kernel32\Interop.VirtualAlloc.cs
Interop\Windows\Kernel32\Interop.VirtualQuery.cs
Interop\Windows\NtDll\Interop.NtQuerySystemInformation.cs
Interop\Windows\User32\Interop.Constants.cs
Interop\Windows\User32\Interop.SendMessageTimeout.cs
```

In each case we should verify the content is the same (or unify it) then delete the System.Private.CoreLib copies in favor of the common ones, updating src\libraries\System.Private.CoreLib\src\System.Private.CoreLib.Shared.projitems and src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj to match

@jkotas I thought we had a work item for unifying interop stuff but I couldn't find it. Maybe it was already complete and this is another.

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #1232

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs [hunk 1]
```diff
@@ -369,42 +369,44 @@ public static void SetAttributes(string fullPath, FileAttributes attributes)
             }
         }
 
-        public static void SetCreationTime(string fullPath, DateTimeOffset time, bool asDirectory)
+        // Default values indicate "no change".  Use defaults so that we don't force callsites to be aware of the default values
+        private static unsafe void SetFileTime(
+            string fullPath,
+            bool asDirectory,
+            long creationTime = -1,
```
Comment (danmosemsft) [inline src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs]: I couldn't find documentation of this "-1" behavior.

File: src/libraries/Common/src/Interop/Windows/Kernel32/Interop.FILE_STANDARD_INFO.cs [hunk 2]
```diff
@@ -2,15 +2,12 @@
...
-using System.Runtime.InteropServices;
-
 internal static partial class Interop
 {
     internal static partial class Kernel32
     {
-        [DllImport(Libraries.Kernel32, SetLastError = true)]
-        internal static extern bool GetFileInformationByHandleEx(SafeFileHandle hFile, FILE_INFO_BY_HANDLE_CLASS FileInformationClass, out FILE_STANDARD_INFO lpFileInformation, uint dwBufferSize);
+        // Use for GetFileInformationByHandleEx
+        internal const int FileStandardInfo = 1;
```
Comment (danmosemsft) [inline src/libraries/Common/src/Interop/Windows/Kernel32/Interop.FILE_STANDARD_INFO.cs]: For random constants like this, would it be helpful to say eg `// from FILE_INFO_BY_HANDLE_CLASS`

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs (hunk 1 above)
Comment (jkotas) [reply src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs]: It is what the existing did. I just moved it around.

I am not able the documentation of this behavior either.

Comment (mikedn) [reply src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs]: `SetFileInformationByHandle` doesn't mention it but you can find this little detail mentioned in `SetFileTime`'s documentation: https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfiletime.

Comment (danmosemsft) [reply src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs]: Oh, good!

**Pull request merged**